### ISSUE TITLE
Maximum call stack size exceeded

### ISSUE THREAD

**Issue by Vijayabhaskar96** [microsoft/pyright#1081]
**Describe the bug**
Throws `Maximum call stack size exceeded` when the below situation occurs.

**To Reproduce**
The code works fine until you do anything by calling `.X()`(like `copy()`, `.get("key")` or .whatever()) to a variable that assigns the value of the variable that created the later one in a loop.
Minimal example(doesn't make sense, just for demo):

```python
test = {
    "key": "value"
}

while True:
    a = test
    test = a.copy()
```

**VS Code extension or command-line**
Using VS Code extension.
VS Code Version:1.49.3
Pyright Version:1.1.76

**Additional context**
This is what it says at the start, after that even if I hover the mouse over any variable or type a single letter it prints almost same error again...
[Info  - 11:17:09 pm] Pyright language server 1.1.76 starting
[Info  - 11:17:09 pm] Server root directory: c:\Users\Vijayabhaskar J\.vscode\extensions\ms-pyright.pyright-1.1.76\dist
Received pythonPath from Python extension: F:\WinPy3710\python-3.7.1.amd64\python.exe
[Error - 11:17:09 pm] stubPath typings is not a valid directory.
[Error - 11:17:11 pm] An internal error occurred while type checking file "f:\WinPy3710\notebooks\project\hello.py": RangeError: Maximum call stack size exceeded
    at Object.y [as specializeType] (c:\Users\Vijayabhaskar J\.vscode\extensions\ms-pyright.pyright-1.1.76\dist\server.js:2:642991)
    at Pe (c:\Users\Vijayabhaskar J\.vscode\extensions\ms-pyright.pyright-1.1.76\dist\server.js:2:496711)
    at Pe (c:\Users\Vijayabhaskar J\.vscode\extensions\ms-pyright.pyright-1.1.76\dist\server.js:2:496698)
    at Pe (c:\Users\Vijayabhaskar J\.vscode\extensions\ms-pyright.pyright-1.1.76\dist\server.js:2:496698)
    at Pe (c:\Users\Vijayabhaskar J\.vscode\extensions\ms-pyright.pyright-1.1.76\dist\server.js:2:496698)
    at Pe (c:\Users\Vijayabhaskar J\.vscode\extensions\ms-pyright.pyright-1.1.76\dist\server.js:2:496698)
    at Pe (c:\Users\Vijayabhaskar J\.vscode\extensions\ms-pyright.pyright-1.1.76\dist\server.js:2:496698)
    at Pe (c:\Users\Vijayabhaskar J\.vscode\extensions\ms-pyright.pyright-1.1.76\dist\server.js:2:496698)
    at Pe (c:\Users\Vijayabhaskar J\.vscode\extensions\ms-pyright.pyright-1.1.76\dist\server.js:2:496698)
    at Pe (c:\Users\Vijayabhaskar J\.vscode\extensions\ms-pyright.pyright-1.1.76\dist\server.js:2:496698)

**Comment by erictraut** [comment]
Thanks for finding a repro! That was very helpful!

I've found and fixed the problem. It was a regression that I introduced when I added support for recursive type aliases. I've added your example as a unit test to prevent a similar regression in the future. The fix will be in the next version of pyright and pylance.

**Comment by erictraut** [comment]
This is now addressed in Pyright 1.1.77, which I just published. It will also be included in the next release of Pylance.
